src/console.cpp

```diff
@@ -169,6 +170,27 @@ void	toggleConsoleDrop()
 	}
 }
 
+bool addConsoleMessageDebounced(const char * Text, CONSOLE_TEXT_JUSTIFICATION jusType, SDWORD player, const DEBOUNCED_MESSAGE & message, bool team, UDWORD duration)
+{
+	// Messages are debounced individually - one debounced message won't stop a different one from appearing
+	static std::map<const DEBOUNCED_MESSAGE *, std::chrono::milliseconds> nextAllowedMessageTimes;
+
+	const std::chrono::milliseconds DEBOUNCE_TIME(message.debounceTime);
+	const std::chrono::milliseconds now = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now().time_since_epoch());
```

Comment:
Instead of using `time_since_epoch()`, it may make more sense to store and compare the `time_point`s directly. 

i.e. Store the last `time_point` (`std::chrono::steady_clock::now()`) for when a debounced message was allowed through, and compute the difference between `std::chrono::steady_clock::now()` and the last value (with appropriate `duration_cast` to milliseconds) to compare versus the `DEBOUNCE_TIME`.